### ISSUE TITLE
[cmake / simprod-scripts] Running make production-tarball; make tarball doesn't make libgsl.so in simulation trunk (Trac #1049)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1043]
Tested this a couple times with trunk of simulation. Running make install_tool_libs will install it, but shouldn't be necessary

Migrated from https://code.icecube.wisc.edu/ticket/1049
```json
{
    "status": "closed", 
    "changetime": "2019-01-11T23:56:01", 
    "description": "Tested this a couple times with trunk of simulation. Running make install_tool_libs will install it, but shouldn't be necessary", 
    "reporter": "mamday", 
    "cc": "david.schultz", 
    "resolution": "fixed", 
    "_ts": "1547250961636294", 
    "component": "cmake", 
    "summary": "[cmake / simprod-scripts] Running make production-tarball; make tarball doesn't make libgsl.so in simulation trunk", 
    "priority": "normal", 
    "keywords": "tarball cmake simprod-scripts", 
    "time": "2015-07-14T18:46:13", 
    "milestone": "", 
    "owner": "nega", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20150714T18:47:33: 
* nega commented:

> double check the target dependencies. when i looked earlier, everything seemed kosher.


* nega changed reporter from "icecube" to "mamday"
* nega changed status from "new" to "accepted"
* nega changed keywords from "" to "tarball"

**Comment by icecube-trac** [comment]
Trac update at 20150716T20:39:49: melanie.day commented:

> I confirmed that make tarball does something different in trunk than it did in the previous release, in that it is not making tarball ROOTSYS and tarball-install-tools. Maybe there is some small setting difference?

**Comment by icecube-trac** [comment]
Trac update at 20150720T21:23:07: melanie.day commented:

> Seems like the problem is in toplevel.cmake? It looks for CVMFS and if the build is not using system packages then it does not add a dependency for tarball-install-tools. That seems like it should be logical, as you would not need the tools in a CVMFS environment. However some things in the tarball are still looking for libraries in the tools directory (like libgsl.so) and it will complain if they are not there (even though they exist in CVMFS). So maybe this is a more complicated problem than I first thought...

**Comment by icecube-trac** [comment]
Trac update at 20150723T22:48:09: 
* nega changed keywords from "tarball" to "tarball cmake simprod-scripts"
* nega commented:

> Ok, so the dynamic linking issue that `make production-tarball` i suppose was designed to fix just comes down making sure the dynamic linker has the correct library path (LD_LIBRARY_PATH).
> 
> in the tarballs this is handled with the LD_LBRARY_PATH set in cvmfs's `setup.sh` which the tarballs `env-shell.sh` inherits, and adds to. neither one have the paths for `gsl`, `GEANT`, or `GENIE`. Things work in a "non-cvmfs tarball" because (most of) the tools get copied in to a "tools" directory, and this is accounted for in the `env-shell.sh`.
> 
> so, temporary fix: add the necessary paths to your LD_LIBRARY_PATH
> 
> permanent fix: be more creative about adding paths LD_LIBRARY_PATH in env-shell.sh. I don't think they should necessarily be added to cvmfs's `setup.sh` but i'll talk to David and get his opinion.
> 
> This should eliminate the need for the `production-tarball` target set in `simprod-scripts`.


* nega changed summary from "Running make production-tarball; make tarball doesn't make libgsl.so in simulation trunk" to "[cmake / simprod-scripts] Running make production-tarball; make tarball doesn't make libgsl.so in simulation trunk"
* nega changed cc from "" to "david.schultz"

**Comment by icecube-trac** [comment]
Trac update at 20150804T15:51:57: 
* nega changed resolution from "" to "fixed"
* nega commented:

> In changeset 2189:
> 
> ```text
> #!CommitTicketReference repository="IceTray" revision="2189"
> fix LD_LIBRARY_PATH for lightweight (cvmfs) tarballs
> 
> versioned installs of other libraries may need to be included in the future
> 
> closes #1043
> ```


* nega changed status from "accepted" to "closed"

**Comment by icecube-trac** [comment]
Trac update at 20190111T23:56:01: nwhitehorn commented:

> In changeset 170067:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="170067"
> fix LD_LIBRARY_PATH for lightweight (cvmfs) tarballs
> 
> versioned installs of other libraries may need to be included in the future
> 
> closes #1043
> ```
